internal/consensus/state.go

```diff
@@ -273,6 +275,23 @@ func (cs *State) SetPrivValidator(priv types.PrivValidator) {
 
 	cs.privValidator = priv
 
+	if priv != nil {
+		t := reflect.TypeOf(priv).Name()
```

Comment:
why do we need to to use reflection here? can't we just use a normal type switch?